fs/remote/resolver.go

```diff
@@ -154,14 +154,14 @@ func (r *Resolver) resolveFetcher(ctx context.Context, hosts source.RegistryHost
 	}
 
 	log.G(ctx).WithError(handlersErr).WithField("ref", refspec.String()).WithField("digest", desc.Digest).Debugf("using default handler")
-	hf, size, err := newHTTPFetcher(ctx, fc)
+	hf, err := newHTTPFetcher(ctx, fc)
 	if err != nil {
 		return nil, 0, err
 	}
 	if blobConfig.ForceSingleRangeMode {
 		hf.singleRangeMode()
 	}
-	return hf, size, err
+	return hf, desc.Size, err
```

Comment:
just a quick question. have you verified, that `desc.Size` contains non-zero value?

Reply:
please don't mind, I think I found the answer in the source.go